Clockwork Cult rework

KeRSedChaplain opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Makes Clockwork Cult less of a clusterfuck of plasma flooding and Gygax rushing (though I know the latter won't be stopped regardless)
This is finally in somewhat of a testmergable state, everything seems to be working on local.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Reebe was a garbage attempt at making clockwork cult unique from Blood Cult, but hopefully, this attempt will be able to capture what made both component-based clock cult AND Reebe clock cult fun.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: The belligerent scripture and a brass multitool, and a new marauder variant which act similar to holoparasites/guardian spirits.
del: Removed the abductor teleport consoles they get, removes abscond for the time being as I've not seen much use for it other than just spamming it and hoping you end up in the armory.
tweak: moved around scriptures to make the cult work better as being based around the station, makes the Ark scream more often and work as a summonable object, clockwork armor now has a flat 0 defense up to 10 instead of negatives against laser damage. Makes the Ark work better in a station based setting, as well as the Heralds beacon in case It works for the mode.
soundadd: added powerloaderstep.ogg for Neovgre
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DeltaFire15 committed:
First of many changes

Adds the possibility for stationside clockwork obelisks to create stable gateways with unlimited uses, linking between the station and reebe. The gateway scripture no longer can teleport cultists from or to reebe. Spatial gateway from obelisks still works, but it is FAR less power efficient.
These gateways are closed via either hitting them with a nullrod, destroying their obelisk, or a cultist initiating their close sequence via a 80ds doafter with the slab used on the gateway.

DeltaFire15 committed:
little thing

Commit:
Adds things to help the cult become station based

Adds (or re-adds I guess) the old spells and stuctures from parts based clock cult, soon to be meant for construction/power generation as well as possible QoL things

Commit:
fucking forgot the defines

Commit:
Initial changes to get this to github for public reviewing

-Removes the Reebe z level and puts it on the cc z level
-Replaces the Neovgre mech stepsound
-Moves spacial gateway into drivers so the cult isn't stuck stationside
-Adds the Belligerant scripture, which applies the Belligerant status effect to non-clankers in range
-Various other changes currently commented out for later work on them.
-removes the abductor teleport cameras from Reebe
-adds a Brass multitool, essentially just an advanced multitool with a special sprite.
-makes clockcult armor have 0 laser defense normally and 10 defense when the ark is almost done summoning Ratvar instead of -25 --> -15
-Makes spear recalling worded to support the addition of new weapons to summon
-Adds new structures for power generation and shuttle delaying
-Other changes I've probably forgotton, but are too lazy to scrounge through the code at 12:43 AM

DeltaFire15 committed:
Updates stable gateways

Changes some stuff around, makes gateways automatically stable if they are a reebe-station obelisk-obelisk gateway

DeltaFire15 committed:
oop

DeltaFire15 committed:
stealthy cogging

APC locks after cogging, drain gets averted by prioritized powerallocation

Commit:
Forgot the 'advanced' prefix here, since thats what it is, essentially

Commit:
ahaha nice I deleted the line that gives them the slab

Commit:
removes the commented out Geis and Cache summon scriptures

Geis won't really work well how it is currently, and Caches could be added in the future as a form of storage/delivery system or power generation.

Commit:
Reebe as its own z is just easier to manage, so its back to being one

Sorry maxcappers and BoH bombers, your cheap tactics will no longer work if you manage to sneak into the void

Commit:
the ark now screams more

halves the time between screams so that cultists know its in trouble. From 60 seconds to 30 seconds.

Commit:
Reebe should probably initialize with the mode again, huh.

Commit:
removes the still unused caches and daemons.

apparently I didn't /actually/ remove the structures fully but here's the rest of them. Oh well, RIP old clockcult buildings, maybe someone will have a use for you in the future.

Commit:
replaces the abscond default quickbind to gateway default quickbind

Commit:
minor Reebe edits

Basically just makes sure all the component decorations in Reebe can't be used for a varying amount of starter power

KeRSedChaplain committed:
Merge pull request #4 from DeltaFire15/COGGERS

Stablegateway stuff from Deltafire but I put it in my own repo

Commit:
scripture changes and new marauder variant

The Ark summoning scripture is being temporarily moved to applications until I can fix it not showing up in a new UI tab, as well as adds the Clockwork Guardian, a marauder variant that acts similar to holoparasites/guardian spirits in the sense that they're linked to a host. slower than normal marauders but with more health, they can automatically block and counter any melee attack, and heal their host up to a small amount if the host is heavily injured (65% health)

Commit:
same as last commit but I actually stage all the right files

Commit:
makes various structures work better for a station based mode

Also updates scripture descriptions to not be from the old version of the mode

Commit:
Stable gateways and integration cog QoL by DeltaFire

APC interfaces now lock after being cogged, and Reebe can only be accessed by stabilizing/linking an obelisk with the one in reebe

silicons committed:
please

silicons committed:
Merge remote-tracking branch 'citadel/master' into 12742

Commit:
updates the UI so it works with TGUI3

Commit:
Fixes the Arks looping issue, and gives Ratvar new words when fighting

Also removes the grace period variable since the ark activates immediately now /shrug

Commit:
Revert "Makes Ratvar spawn above the Ark"

This reverts commit e4ae6134349b4fdcebec93a18de3df72a02a2af8.

Commit:
Makes Ratvar spawn above the Ark

Commit:
Revert "Makes Ratvar spawn above the Ark"

This reverts commit e4ae6134349b4fdcebec93a18de3df72a02a2af8.
and GitKraken being laggy made it happen twice hrrng

Commit:
Revert "makes Travis not yell at marauders I don't know they worked fine /shrug"

This reverts commit edd68c5174f1c3315373b52f59472b5500af7f19.

Commit:
makes Travis not yell at marauders I don't know they worked fine /shrug

Commit:
Revert "/probably/ gets Travis to stop shouting about guardians"

This reverts commit be9723678e7c1f88817ac6f615aea6e0222b6fea.

Commit:
/probably/ gets Travis to stop shouting about guardians

Commit:
God I hate travis, will this appease you?

timothyteakettle commented on code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm:
```diff
@@ -124,3 +124,435 @@
+	return ..()
+
+/mob/living/simple_animal/hostile/clockwork/marauder/guardian/hitby(atom/movable/AM, skipcatch, hitpush, blockedatom/movable/AM, datum/thrownthing/throwingdatum)
+	if(blockOrCounter(null, AM))
```
it should be blocked not blockedatom/movable/AM
you already defined atom/movable/AM

Commit:
My illiteracy knows no bounds

Commit:
probably gets travis to stop yelling about the new reebe

Commit:
Buffs some of the available scriptures for clock cult borgs

engineering borgs get a focus on the structures
combat oriented borgs get more combat spells IE: belligerent and volt blaster
mediborgs get mending mantra
comments out the janitor slab things, to mirror the janiborg being commented out

Commit:
increases the penalty for off-station scripture recital to x3 cost/time

Commit:
description edits/Ratvar calling you out for trying to make another Ark

makes the Ark's description fit for a station based setting, and makes Ratvar have a nicer tone when talking to someone trying to summon another ark after he's spawned already

Commit:
Makes the god fight cooler by adding new quotes and some sound changes

DeltaFire15 committed:
modifies the ark to not immediately activate and purges some todos

Tin

DeltaFire15 committed:
modifies the ark, aswell as changing a tiny bit of stuff for stable gateways

The ark no longer immediately activates when summoned. Instead, an cultist needs to activate it. Once on, it cannot be turned off again save by destroying it.

Also, the stable-gateway deactivation doafter now has an actual beam-animation.

KeRSedChaplain committed:
Merge pull request #7 from DeltaFire15/clockrework-arksequence

Clockrework arksequence-done by Deltafire

Commit:
Revert "Revert "THIS COMMIT IS PURELY SO IT WORKS WITH KEVINS CLICKDELAYS #12834""

This reverts commit 4b876ef9fb1049bd7e55db5326fa2908adc8799d.
yo hol up I gotta see if the confliction is fixed first

Commit:
Revert "THIS COMMIT IS PURELY SO IT WORKS WITH KEVINS CLICKDELAYS #12834"

This reverts commit 7e1cab15ae7e8544a9e299ca5134c14e138086fd.

Commit:
THIS COMMIT IS PURELY SO IT WORKS WITH KEVINS CLICKDELAYS #12834

REVERT IF CLICKDELAYS ARE UN-TESTMERGED

Commit:
Revert "Revert "Revert "THIS COMMIT IS PURELY SO IT WORKS WITH KEVINS CLICKDELAYS #12834"""

This reverts commit 0e6fe357da0f8ca372e3218a894b27f23d75ef72.

Commit:
I didn't get to see if it'd still conflict with the clickdelay testmerge

ugghh I shouldn't be coding when my second brain cell falls asleep but hey at least the change is incredibly small

Commit:
still conflicts with clickdelays bluh whatever I'll fix it if its merged

DeltaFire15 committed:
Updates the clockie delay structure

Prolonging prisms exist again. They can be built application-tier and take one minute to delay the shuttle by two minutes (So, basically, you end up with one minute more on the timer before it started). It costs 2000 power for the initial use, and 1000 additional power per time it was already used. Oh, and it creates a reeeaaally obvious effect pointing anyone towards the structure. So, good luck defending it.
Also fixes up some scripture sorting but that's minor.

KeRSedChaplain committed:
Merge pull request #9 from DeltaFire15/prolonging-prism-update

Updates the clockie delay structure

Commit:
various fixes to make the PR not conflict with clickdelays probably

Commit:
updates clockwork armor /what is magic defense even for i don't get it/

silicons committed:
Merge remote-tracking branch 'citadel/master' into 12742

silicons committed:
fix

silicons committed:
revert cc

silicons committed:
Merge remote-tracking branch 'citadel/master' into 12742

DeltaFire15 committed:
Fixes the clockcult gamemode

Servants no longer spawn / stay on the start screen and have no role (haha OOPS). Things SHOULD work now, at least on local with debug mode on, I give no guarantees.

KeRSedChaplain committed:
Merge pull request #11 from DeltaFire15/HAHA-FUCK

Fixes the clockcult gamemode -Deltafire

Commit:
updates the credits bit of clockwork cult

Commit:
fixes the inhand sprite for clockwork slabs

Commit:
Changes the death sound for Ratvar to have more oomph behind it

Commit:
Turns out I was the one screaming all along

Commit:
fixes the invisible slabs and buffs a couple scripture casting times

Commit:
HAHAHAHA NOTHING TO SEE HERE

THIS VARIABLE HAS ALWAYS BEEN THERE WHAT DO YOU MEAN HAHAHAHAHA

Commit:
changes Neovgre's sounds again
